backend/src/user/password-cryptographer/password-cryptographer.ts

```diff
@@ -0,0 +1,46 @@
+import * as bcrypt from 'bcrypt-nodejs';
+import {Component} from '@nestjs/common';
+import {PasswordCryptographerService} from './password-cryptographer.interface';
+
+@Component()
+export class PasswordCryptographerServiceImpl implements PasswordCryptographerService {
+
+  doHash (plaintextPassword: string): Promise<string> {
+
+    return new Promise((resolve, reject) => {
+
+      bcrypt.genSalt(this.saltRounds(), (error, salt) => {
+        bcrypt.hash(plaintextPassword, salt, null, (err, hash) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(hash);
+          }
+        });
+      });
+
+    });
+
+  }
+
+  doCompare (plaintextPassword, hash): Promise<boolean> {
+
+    return new Promise((resolve, reject) => {
+
+      bcrypt.compare(plaintextPassword, hash, (err, res) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(res);
+        }
+      });
+
+    });
+
+  }
+
+  private saltRounds() {
```

Comment:
this should be just a field, not a method with hardcoded value

Reply:
ok